samples/crypto/kmu_usage_nrf54l/src/main.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/logging/log.h>
+#include <stdio.h>
+#include <psa/crypto.h>
+#include <psa/crypto_extra.h>
+
+#include <cracen_psa_kmu.h>
+
+#ifdef CONFIG_BUILD_WITH_TFM
+#include <tfm_ns_interface.h>
+#endif
+
+#include "main.h"
+#include "key_operations.h"
+#include "init.h"
+
+#define APP_SUCCESS	    (0)
+#define APP_ERROR	    (-1)
+#define APP_SUCCESS_MESSAGE "Example finished successfully!"
+#define APP_ERROR_MESSAGE   "Example exited with error!"
+
+LOG_MODULE_REGISTER(kmu_usage_nrf54l, LOG_LEVEL_DBG);
+
+/* ====================================================================== */
+/*			Global variables/defines for the KMU usage example			  */
+
+#define SAMPLE_MAX_KEY_COUNT (4)
+
+static sample_key_entry_t m_sample_keys[SAMPLE_MAX_KEY_COUNT] = {
+	/* Pre-provisioned keys.
+	 * These keys remain in the KMU between MCU resets.
+	 */
+	{
+		/* The key requires 1 KMU slot */
+		.key_id = PSA_KEY_HANDLE_FROM_CRACEN_KMU_SLOT(CRACEN_KMU_KEY_USAGE_SCHEME_PROTECTED,
+						       PSA_KEY_ID_USER_MIN),
+		.supported_operations = {
+			.gen_key_cb = NULL,
+			.use_key_cb = key_operations_use_aes_key,
+		}
+	},
+	{
+		/* The key requires 2 KMU slots */
+		.key_id = PSA_KEY_HANDLE_FROM_CRACEN_KMU_SLOT(CRACEN_KMU_KEY_USAGE_SCHEME_RAW,
+						       PSA_KEY_ID_USER_MIN + 1),
+		.supported_operations = {
+			.gen_key_cb = NULL,
+			.use_key_cb = key_operations_use_eddsa_key_pair,
+		}
+	},
+
+	/* Generated keys.
+	 * These keys will be erased from the KMU after MCU reset.
+	 */
+	{
+		/* The key requires 1 KMU slot + 2 additional slots
+		 *are required since the key is encrypted
+		 */
+		.key_id = PSA_KEY_HANDLE_FROM_CRACEN_KMU_SLOT(CRACEN_KMU_KEY_USAGE_SCHEME_ENCRYPTED,
+						       PSA_KEY_ID_USER_MIN + 3),
+		.supported_operations = {
+			.gen_key_cb = key_operations_generate_aes_key,
+			.use_key_cb = key_operations_use_aes_key,
+		}
+	},
+	{
+		/* The key requires 2 KMU slots */
+		.key_id = PSA_KEY_HANDLE_FROM_CRACEN_KMU_SLOT(CRACEN_KMU_KEY_USAGE_SCHEME_RAW,
+						       PSA_KEY_ID_USER_MIN + 6),
+		.supported_operations = {
+			.gen_key_cb = key_operations_generate_ecdsa_key_pair,
+			.use_key_cb = key_operations_use_ecdsa_key_pair,
+		}
+	},
+};
+
+/* ====================================================================== */
+
+int crypto_init(void)
+{
+	psa_status_t status;
+
+#if defined(CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK) ||                            \
+	defined(CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_KEY_PROVIDER_HUK_LIBRARY)
+	write_huk();
+#endif
```

Comment:
No need for this on nRF54L15, which is the only supported board for this sample. We use the IKG, not the HUK and the KMU does not require either